src/SingleDelimiterRule.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Deg540\StringCalculatorPHP;
+
+class SingleDelimiterRule implements IDelimitersRule
+{
+    public function extractDelimiters(string $string): ProcessedString
+    {
+        $processedString = new ProcessedString();
+        $delimiters = new Delimiters();
+        if ($this->isSingleCharacterDelimiter($string)) {
+            $delimiters->delete();
+            $delimiters->add($this->extractDelimiter($string));
+            $processedString->setDelimiters($delimiters);
+            $processedString->processNumbers($this->extractNumbers($string));
+        }
+
+        return $processedString;
+    }
```

Comment:
We could process the string directly in each rule and return the numbers to operate?
```suggestion
    /**
     * @return int[]
     */
    public function processString(string $string): array
    {
        if (!$this->isSingleCharacterDelimiter($string)) {
            return [];
        }

        $delimiter = $this->extractDelimiter($string);

        return $this->extractNumbers($string, $delimiter);
    }
```